app_flutter/lib/task_icon.dart

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+import 'package:cocoon_service/protos.dart' show Task;
+
+/// Header icon for all [Task] that map to the same [task.stageName] and [task.name].
+///
+/// Intended to be used in the first row of [StatusGrid].
+/// Shows an icon based on [task.stageName], defaulting to [Icons.help] if it can't be mapped.
+/// On tap, shows [task.name].
+class TaskIcon extends StatelessWidget {
+  const TaskIcon({Key key, @required this.task}) : super(key: key);
+
+  /// [Task] to get information from.
+  final Task task;
+
+  /// [stageName] that maps to StageName enums.
+  // TODO(chillers): Remove these and use StageName enum when available. https://github.com/flutter/cocoon/issues/441
+  static const String stageCirrus = 'cirrus';
+  static const String stageLuci = 'chromebot';
+  static const String stageDevicelab = 'devicelab';
+  static const String stageDevicelabWin = 'devicelab_win';
+  static const String stageDevicelabIOs = 'devicelab_ios';
+
+  /// A lookup table for matching [stageName] to [Image].
+  ///
+  /// [stageName] is based on the backend.
+  static Map<String, Image> stageIcons = <String, Image>{
```

Comment:
Should be final

Reply:
Can it be made final? I get issues from the constructor saying it can't be final since Image.asset is not const.

Reply:
`static final Map<String, Image> stageIcons = ...` should be fine

Reply:
Thanks, that fixed it